File: src/main/java/com/assetmanagement/notification/service/NotificationService.java
package com.assetmanagement.notification.service;

import com.assetmanagement.notification.dto.NotificationResponse;
import com.assetmanagement.notification.enums.NotificationType;

import java.util.List;

public interface NotificationService {

    /**
     * Create and persist a notification for a specific recipient (by employee ID).
     */
    void createNotification(Long recipientId, String title, String message,
                            NotificationType type, Long referenceId, String referenceType);

    /**
     * Retrieve all notifications for a given employee.
     */
    List<NotificationResponse> getNotificationsForEmployee(Long employeeId);

    /**
     * Retrieve only unread notifications for a given employee.
     */
    List<NotificationResponse> getUnreadNotificationsForEmployee(Long employeeId);

    /**
     * Count unread notifications for a given employee.
     */
    long countUnread(Long employeeId);

    /**
     * Mark a single notification as read.
     */
    NotificationResponse markAsRead(Long notificationId);

    /**
     * Mark all notifications as read for a given employee.
     */
    int markAllAsRead(Long employeeId);
}

File: frontend/src/api/notificationService.ts
import api from "./axios";
import type { NotificationResponse } from "../types/notification";

export const notificationService = {
  getAll: async (employeeId: number): Promise<NotificationResponse[]> => {
    const res = await api.get(`/notifications/employee/${employeeId}`);
    return res.data;
  },

  getUnread: async (employeeId: number): Promise<NotificationResponse[]> => {
    const res = await api.get(`/notifications/employee/${employeeId}/unread`);
    return res.data;
  },

  getUnreadCount: async (employeeId: number): Promise<{ unreadCount: number }> => {
    const res = await api.get(`/notifications/employee/${employeeId}/count-unread`);
    return res.data;
  },

  markAsRead: async (notificationId: number): Promise<NotificationResponse> => {
    const res = await api.put(`/notifications/${notificationId}/read`);
    return res.data;
  },

  markAllAsRead: async (employeeId: number): Promise<{ markedRead: number }> => {
    const res = await api.put(`/notifications/employee/${employeeId}/read-all`);
    return res.data;
  }
};

